docs/porting-to-ROS2.md

```diff
@@ -174,14 +248,59 @@ You could do therefore try setting up a dedicated thread, but you could also use
     tf_buffer_.setCreateTimerInterface(cti);
     ...
     // In the function processing data
-    tf_buffer_.waitForTransform(a, b, msg_time, std::chrono::milliseconds(1000), [this](const std::shared_future<geometry_msgs::msg::TransformStamped>& tf){
+    auto tf_future = tf_buffer_.waitForTransform(a, b, msg_time, std::chrono::milliseconds(1000), [this](const std::shared_future<geometry_msgs::msg::TransformStamped>& tf){
         // Here the future is available
     });
 
 The callback will always be called, but only after some time: when the transform becomes available or when the timeout is reached. In the latter case, if the transform is not ready yet, calling `.get()` on the future will throw a `tf2::TimeoutException`.
 
-The `waitForTransform()` function will return immediately and also return a future. However, calling `.get()` or `.wait()` on that future does not respect the timeout. That is, it will wait however long it takes until a transform arrives and never throw an exception.
+But there is a better way that doesn't need exceptions. The `waitForTransform()` function will
+return immediately and also return a future. However, calling `.get()` or `.wait()` on that future
+does not respect the timeout. That is, it will wait however long it takes until a transform arrives
+and never throw an exception. So, to wait on the returned future for a specified time,
+
+    tf_future.wait_for(timeout)
+
+Putting all pieces together, to block until a specific transform is available, create a separate method and call it at the end of the constructor to ensure that all members are properly initialized first
+
+```c++
+MPCFollower::MPCFollower()
+{
+...
+blockUntilVehiclePositionAvailable(tf2::durationFromSec(5.0));
+}
+
+void MPCFollower::blockUntilVehiclePositionAvailable(const tf2::Duration & timeout)
+{
+  auto cti = std::make_shared<tf2_ros::CreateTimerROS>(
+    this->get_node_base_interface(), this->get_node_timers_interface());
+  tf_buffer_.setCreateTimerInterface(cti);
+
+  while (rclcpp::ok()) {
+    static constexpr auto input = "map", output = "base_link";
+    auto tf_future = tf_buffer_.waitForTransform(
+      input, output, tf2::TimePointZero, tf2::durationFromSec(0.0), [](auto &) {});
+    const auto status = tf_future.wait_for(timeout);
+    if (status == std::future_status::ready) {
+      break;
+    } else {
+      RCLCPP_INFO(
+        get_logger(), "waiting another %d seconds for %s->%s transform",
+        std::chrono::duration_cast<std::chrono::seconds>(timeout).count(), input, output);
+    }
+  }
+}
+
+```
 
+### Shared pointers
```

Comment:
Does this scenario tend to come up a lot when porting? If yes, where?

Reply:
I don't know if it happens often but it happened to me once [here](https://github.com/tier4/Pilot.Auto/pull/52/files?file-filters%5B%5D=.cfg&file-filters%5B%5D=.cpp&file-filters%5B%5D=.py&file-filters%5B%5D=.test&file-filters%5B%5D=.txt&file-filters%5B%5D=.xml&file-filters%5B%5D=.yaml&file-filters%5B%5D=No+extension#r510717033)

Reply:
I was just wondering whether it's a general C++ tip or specific to ROS1->ROS2 porting. I'll leave it up to you.